### PR TITLE
dqtool: add option for truncating (compacting) abandoned disk-buffers

### PR DESCRIPTION
**Pull request by MrAnno**

### CODE REVIEW

**Review by alltilla — APPROVED** [review]

File: modules/diskq/dqtool.c [hunk 1]
```diff
@@ -629,6 +665,7 @@ static struct
   { "info", info_options, "Print infos about the given disk queue file", dqtool_info },
   { "relocate", relocate_options, "Relocate(rename) diskq file. Note that this option modifies the persist file.", dqtool_relocate },
   { "assign", assign_options, "Assign diskq file to the given persist file with the given persist name.", dqtool_assign },
+  { "truncate", truncate_options, "Truncate unused space in abandoned disk queues", dqtool_truncate },
```
[the remark this answers is not in the document]
Comment (MrAnno) [reply modules/diskq/dqtool.c:668]: My first version had the name "compact", but I realized we have a compaction feature in disk-buffer, so the name collides. I also realized that using this command is not recommended on "non-abandoned" buffers, and until we make some kind of programmatic validation, the name is better to sound as destructive as possible.

File: modules/diskq/dqtool.c [hunk 2]
```diff
@@ -142,6 +149,36 @@ open_queue(char *filename, LogQueue **lq, DiskQueueOptions *options)
...
+
+      printf("Truncating disk-buffer %s\n", argv[i]);
+      options.truncate_size_ratio = 0;
+
+      if (!open_queue(argv[i], &lq, &options, FALSE))
+        continue;
+
+      gboolean persistent;
+      log_queue_disk_stop(lq, &persistent);
+      log_queue_unref(lq);
```
[the remark this answers is not in the document]
Comment (MrAnno) [reply modules/diskq/dqtool.c:176]: Good idea, I'm adding it.

[the remark this answers is not in the document]
Comment (MrAnno) [reply modules/diskq/dqtool.c:176]:
```
Disk-buffer var/syslog-ng-00001.qf has been truncated, reclaimed space: 1.862641 GiB
```

**Comment by MrAnno** [comment]
@HofiOne Something is broken in the macOS 14 job. Can you take a look, please?

**Comment by MrAnno** [comment]
This is also red only because of the macOS failure.

**Comment by HofiOne** [comment]
> @HofiOne Something is broken in the macOS 14 job. Can you take a look, please?

#4878 adds a solution for this, it's a bit hacky, but actually, that was the best I could find for this multiple-layered issue

**Pull request merged**